### PR TITLE
W-10876269: Add exceptions records to spans and error data to on error handlers

### PR DESCRIPTION
**Pull request by IvanAndresFritzler**

### CODE REVIEW

File: core/src/main/java/org/mule/runtime/core/internal/profiling/tracing/event/span/InternalSpan.java [hunk 1]
```diff
@@ -33,6 +35,12 @@ public interface InternalSpan extends Span {
    */
   void end();
 
+
+  /**
+   * Ends the span, previously recording the provided error.
```
Comment (fsgonz) [inline core/src/main/java/org/mule/runtime/core/internal/profiling/tracing/event/span/InternalSpan.java:40]: Why we should end the span when an error is registered? I would think we should

Comment (fsgonz) [reply core/src/main/java/org/mule/runtime/core/internal/profiling/tracing/event/span/InternalSpan.java:40]: I think this is only a jdoc comment.

File: core/src/main/java/org/mule/runtime/core/internal/profiling/tracing/export/OpenTelemetrySpanExporter.java [hunk 2]
```diff
@@ -67,17 +80,31 @@ public <T> T visit(InternalSpanExporterVisitor<T> internalSpanExporterVisitor) {
...
+  }
+
+  private void recordSpanException(SpanError spanError) {
+    Attributes errorAttributes = Attributes.of(
+                                               EXCEPTION_TYPE_KEY, spanError.getError().getErrorType().toString(),
+                                               EXCEPTION_MESSAGE_KEY, spanError.getError().getDescription(),
+                                               EXCEPTION_STACK_TRACE_KEY,
+                                               InternalSpanError.getInternalSpanError(spanError).getErrorStacktrace().toString(),
+                                               EXCEPTION_ESCAPED_KEY, spanError.isEscapingSpan());
+    openTelemetrySpan.addEvent("exception", errorAttributes);
```
Comment (fsgonz) [inline core/src/main/java/org/mule/runtime/core/internal/profiling/tracing/export/OpenTelemetrySpanExporter.java:94]: add this to the constants too.
Add a TODO to this ticket: W-11610439: Track if the semconv dependency (alpha) should be added.
To remember to verify if it convenient to add these constants from the semconv project

File: core/src/main/java/org/mule/runtime/core/internal/profiling/tracing/event/span/InternalSpan.java (hunk 1 above)
Comment (IvanAndresFritzler) [reply core/src/main/java/org/mule/runtime/core/internal/profiling/tracing/event/span/InternalSpan.java:40]: It is not ending the Span, it was just the JDoc.

**Review by fsgonz — APPROVED** [review]

**Pull request merged**